disnake/state.py

```diff
@@ -819,7 +832,11 @@ def parse_message_delete_bulk(self, data: gateway.MessageDeleteBulkEvent) -> Non
         guild = self._get_guild(raw.guild_id)
         thread = guild and guild.get_thread(raw.channel_id)
         if thread:
-            thread.message_count = max(0, thread.message_count - len(raw.message_ids))
+            to_subtract = len(raw.message_ids)
+            # the initial message isn't counted, and hence shouldn't be subtracted from the count either
```

Comment:
because we don't check the message type here is thread_starter_message, I wonder if we need to check for thread_starter_message above too.

Reply:
It should be safe to assume that if `msg.type == thread_starter_message`, then `msg.id == ch.id`, so the check in `parse_message_create` could possibly be simplified to just `msg.id == ch.id`.
The current implementation in all three event handlers mirrors the client, which checks type/parent/id in `message_create`, and just the id in `message_delete` and `message_delete_bulk`, so I'm not sure - either option should be fine.

Reply:
That's fair. We probably don't need to check thread_starter_message in the message create but I don't know.

Reply:
Nevermind, I mixed up the IDs, this is wrong:

> It should be safe to assume that if `msg.type == thread_starter_message`, then `msg.id == ch.id`

Meaning the message type check is still required for non-forum threads.

Reply:
Ah.